pilot/pkg/model/context.go

```diff
@@ -718,6 +718,9 @@ type NodeMetadata struct {
 	// The istiod address when running ASM Managed Control Plane.
 	CloudrunAddr string `json:"CLOUDRUN_ADDR,omitempty"`
 
+	// Metadata discovery service enablement
```

Comment:
Can you add a bit more ? I assume this is a node metadata that allows an envoy to opt in on using MDS ? Can it also opt out by setting it to false ? Will setting it to false result in the old MX being used - or how does it interact with the 2 global settings ? 

In general I think per-node opt in / opt-out is far better in particular for risky things, so I would prefer if this is the main mechanism (instead of globals).

Reply:
Yes, it's per node enablement. It's off by default, requires an env variable in the sidecar to turn on.